src/com/google/javascript/jscomp/js/es6/util/inherits.js

```diff
@@ -49,25 +49,31 @@
  *
  * @param {!Function} childCtor Child class.
  * @param {!Function} parentCtor Parent class.
+ * @param {boolean} es5Out language out is ES5
  */
-$jscomp.inherits = function(childCtor, parentCtor) {
-  /** @constructor */
-  function tempCtor() {}
-  tempCtor.prototype = parentCtor.prototype;
-  childCtor.superClass_ = parentCtor.prototype;
-  childCtor.prototype = new tempCtor();
-  /** @override */
-    childCtor.prototype.constructor = childCtor;
+$jscomp.inherits = function(childCtor, parentCtor, es5Out) {
+  if (es5Out === true && Object.setPrototypeOf) {
```

Comment:
When I suggested the extra argument for `$jscomp.inherits()` I believed this to be true, which may be false:
1. All browsers that support all ES5 features also support `setPrototypeOf` (or `__proto__` which allows our polyfill to work) despite the fact that those aren't actually defined in ES5.

FWIW, MDN says that IE11 was the earliest IE version to support them. I don't know right off how to determine whether any IE < 11 should be considered ES5 compliant.

Based on that belief I was expecting we would support these scenarios.

1. Developer asks for ES3 output, basically saying we cannot count on `setPrototypeOf` working. In this case we do not try to use `setPrototypeOf()`, because we want to make sure the code we output has the same behavior on all browsers.
2. Developer asks for ES5 output, which is effectively a promise not to run our output on a browser that doesn't have full support for ES5, and by my assumption above, for `setPrototypeOf()`. In this case we use `setPrototypeOf()` and expect it to be there. If it isn't there, we need to let the developer know by treating that as an error.

Do you know whether my assumption is true or false?
Do you have other problems with this plan?
Thanks.